dev/com.ibm.ws.jaxrs.web/src/com/ibm/ws/jaxrs/web/JaxRsInjectionClassListCollaborator.java

```diff
@@ -0,0 +1,265 @@
+/*******************************************************************************
+ * Copyright (c) 2012, 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.jaxrs.web;
+
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import javax.ws.rs.ApplicationPath;
+import javax.ws.rs.ext.Provider;
+
+import org.osgi.service.component.annotations.Component;
+
+import com.ibm.ws.container.service.annocache.AnnotationsBetaHelper;
+import com.ibm.ws.container.service.annotations.WebAnnotations;
+import com.ibm.ws.container.service.app.deploy.WebModuleInfo;
+import com.ibm.wsspi.adaptable.module.Container;
+import com.ibm.wsspi.adaptable.module.NonPersistentCache;
+import com.ibm.wsspi.adaptable.module.UnableToAdaptException;
+import com.ibm.wsspi.anno.info.ClassInfo;
+import com.ibm.wsspi.anno.info.MethodInfo;
+import com.ibm.wsspi.anno.targets.AnnotationTargets_Targets;
+import com.ibm.wsspi.webcontainer.collaborator.WebAppInjectionClassListCollaborator;
+
+/**
+ * A WebAppInjectionClassListCollaborator is required to tell CDI which application classes require injection.
+ */
+@Component(name = "com.ibm.ws.jaxrs.web.JaxRsInjectionClassListCollaborator",
+           service = WebAppInjectionClassListCollaborator.class,
+           immediate = true,
+           property = { "service.vendor=IBM" })
+public class JaxRsInjectionClassListCollaborator implements WebAppInjectionClassListCollaborator {
+
+    /** {@inheritDoc} */
+    @Override
+    public List<String> getInjectionClasses(Container moduleContainer) {
+        System.out.println("JaxRsInjectionClassListCollaborator.getInjectionClasses()");
+
+        try {
+            if (!isWebModule(moduleContainer)) {
+                return Collections.<String> emptyList();
+            }
+            return getJaxRsInjectionClasses(moduleContainer);
+        } catch (UnableToAdaptException e) {
+            return Collections.<String> emptyList();
+        }
+
+    }
+
```

Comment:
This method is copied from JaxRsUtils

Reply:
For the methods that you copied from JaxRsUtils and InjectionUtils, can we remove them?  Or are they called from other places in the code?

Reply:
not sure at this time